monitoring/reporter/expvar/expvars.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package expvar
+
+import (
+	"expvar"
+	"net/http"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/elastic/elastic-agent-libs/transform/typeconv"
+	"github.com/elastic/elastic-agent-shipper/monitoring/reporter"
+)
+
+//Config is the config struct for marshalling whatever we get from the config file
+type Config struct {
+	Enabled bool   `config:"enabled"`
+	Addr    string `config:"address"`
+	Name    string `config:"name"`
+}
+
+// Expvars is the simple manager for the expvars web interface
+type Expvars struct {
+	log     *logp.Logger
+	metrics reporter.QueueMetrics
+	srv     *http.Server
+}
+
+// NewExpvarReporter initializes the expvar interface, and starts the http frontend.
+func NewExpvarReporter(cfg Config) reporter.Reporter {
+	exp := Expvars{
+		log:     logp.L(),
+		srv:     &http.Server{Addr: cfg.Addr},
+		metrics: reporter.QueueMetrics{},
+	}
+	exp.log.Debugf("Starting expvar monitoring...")
+	expvar.Publish(cfg.Name, expvar.Func(exp.format))
+	exp.runFrontend(cfg)
+	return &exp
+}
+
+func (exp Expvars) runFrontend(cfg Config) {
+	srv := &http.Server{Addr: cfg.Addr}
+	go func() {
+		err := srv.ListenAndServe()
+		if err != nil {
+			// Error type isn't happy with %w here
+			exp.log.Errorf("Error starting HTTP expvar server: %s", err)
```

Comment:
You may want to filter out `http.ErrServerClosed` here, which will always happen on shutdown.